### PR TITLE
Add logout button and encourage encryption

### PR DESCRIPTION
**Pull request by panleone**
## Abstract

Add a log out button ( where log out of course means also remove the account from the database) and explicitly ask users to encrypt their wallet on creation/non-view import.
With the new wallet model we want to minimize the number of "first sync" (PR #219) that might take a while. If the wallet is encrypted txs can be saved and fetched on DB (PR for this will come soon), reducing the number of "first sync" to 1.

---

## Testing
- Verify that log out works fine 
- Verify that encryption is asked on wallet creation/import

### CODE REVIEW

File: scripts/database.js [hunk 1]
```diff
@@ -430,4 +430,11 @@ export class Database {
 
         return this.#instances.get(name);
     }
+
+    static async removeIstance() {
```
Comment (JSKitty) [inline scripts/database.js:434]: Typo.
```suggestion
    static async removeInstance() {
```

**Review by JSKitty — CHANGES_REQUESTED** [review]
I think we should call this "Delete Wallet" or "Wipe Data", etc - people will confuse "Log out" as simply having to enter their password and they get back, but this will delete everything: their Contacts, their private key, too dangerous to give to average users.

It should also have a second "Are you sure?", "You will lose all your wallet data" prompt popup after hitting the button, for safety.

**Comment by panleone** [comment]
> I think we should call this "Delete Wallet" or "Wipe Data", etc - people will confuse "Log out" as simply having to enter their password and they get back, but this will delete everything: their Contacts, their private key, too dangerous to give to average users.
> 
> It should also have a second "Are you sure?", "You will lose all your wallet data" prompt popup after hitting the button, for safety.

Agree, I'll implement this later

File: scripts/settings.js [hunk 2]
```diff
@@ -469,7 +469,20 @@ async function setAnalytics(level, fSilent = false) {
...
+            { network: cChainParams.current.name },
+        ])}</b>
+        <br>
+        This will delete all your data, including masternodes contacts and private keys!
+        <br>
+        <br>
+    `,
+    });
+    if (!fContinue) return;
+    Database.removeInstance();
```
Comment (JSKitty) [inline scripts/settings.js:485]: Nit: Should we `await` this? I don't see much harm happening, but I think making sure most DB interactions are async/awaited can save us some future race issues.

**Review by JSKitty — DISMISSED** [review]
**tACK** 2c5695496910acf377b83911556cb6918ed30a47

File: scripts/settings.js [hunk 3]
```diff
@@ -464,6 +465,55 @@ async function setAnalytics(level, fSilent = false) {
...
+ */
+export async function logOut() {
+    const fContinue = await confirmPopup({
+        title: 'Are you sure?',
+        html: `
+        <b>${tr(translation.netSwitchUnsavedWarningSubtitle, [
+            { network: cChainParams.current.name },
+        ])}</b>
+        <br>
+        This will delete all your data, including masternodes contacts and private keys!
```
Comment (Duddino) [inline scripts/settings.js:479]: Translate this

**Review by Duddino — CHANGES_REQUESTED** [review]
Balance is not reset to 0 when deleting a wallet and immediately importing/creating a new one

**Comment by JSKitty** [comment]
Currently the `Encryption` dialog doesn't actually force it - the user can just hit `X` or the backdrop to close it.

We also need to keep in mind that menu is re-used for "Change Password", so removing the ability to close should be optional (we don't want to force them to change password if they hit the button haha).

**Comment by panleone** [comment]
merged with master

File: scripts/database.js [hunk 4]
```diff
@@ -506,4 +506,11 @@ export class Database {
 
         return this.#instances.get(name);
     }
+
+    static async removeInstance() {
+        const name = cChainParams.current.name;
+        const instance = this.#instances.get(name);
+        if (instance) await instance.removeAccount({ publicKey: null });
+        this.#instances.delete(name);
+    }
```
Comment (Duddino) [inline scripts/database.js:515]: This doesn't actually remove the instance. Instead `Database::removeAccount` should be called

File: scripts/settings.js [hunk 5]
```diff
@@ -464,6 +465,55 @@ async function setAnalytics(level, fSilent = false) {
...
+            { network: cChainParams.current.name },
+        ])}</b>
+        <br>
+        ${translation.CONFIRM_POPUP_DELETE_ACCOUNT}
+        <br>
+        <br>
+    `,
+    });
+    if (!fContinue) return;
+    await Database.removeInstance();
```
Comment (Duddino) [inline scripts/settings.js:485]: Call `Database::removeAccount` instaed

File: scripts/database.js (hunk 4 above)
Comment (panleone) [reply scripts/database.js]: nice catch

File: locale/en/translation.js [hunk 6]
```diff
@@ -228,6 +228,7 @@ export default {
     ID: 'ID',
     time: 'Time',
     description: 'Description',
+    accountDeleted: 'Your accoun has been successfully deleted!',
```
Comment (JSKitty) [inline locale/en/translation.js:231]:
```suggestion
    accountDeleted: 'Your account has been successfully deleted!',
```

**Review by JSKitty — APPROVED** [review]
**tACK** besides from minor typo, looking good, simple and a nice feature to have finally.

**Review by JSKitty — APPROVED** [review]
**re-tACK**

**Review by Liquid369 — APPROVED** [review]
tACK 210a7ec73dfbc520ed289d4fbf7af7bf857439f2
jumped in and tested it out, works as intended, but waited for typo fix to submit lol

**Review by Duddino — APPROVED** [review]
tACK

**Pull request merged**